docs/DafnyRef/UserGuide.md

```diff
@@ -571,6 +571,23 @@ Most output from `dafny` is directed to the standard output of the shell invokin
 - Dafny `expect` statements (when they fail) send a message to **standard-out**.
 - Dafny I/O libraries send output explicitly to either **standard-out or standard-error**
 
+### 13.5.5. Project files
+
+Commands on the Dafny CLI that can be passed a Dafny file, can also be passed a Dafny project file. Such a project file may define which Dafny files the project contains, and which Dafny options it uses. The project file must be a [TOML](https://toml.io/en/) file named `dafny.toml`. Here's an example of a Dafny project file:
```

Comment:
It feels a bit weird to be passing the project file directly as just another file, rather than with a flag like `--project`. It also means you can't have multiple project files for a codebase in the same directory, which is a less common scenario but one I have seen in other systems frequently enough.

It also seems valuable to search for a project file automatically just as `dafny server` will. Future improvement or something you're intentionally omitting?

Reply:
> It feels a bit weird to be passing the project file directly as just another file, rather than with a flag like --project. It also means you can't have multiple project files for a codebase in the same directory

Why would it mean that? Passing project files as just other files means that the extension needs to be used for Dafny to determine how to interpret the file. Any `.toml` file is considered a project file. That's fine right?

> It also means you can't have multiple project files for a codebase in the same directory, which is a less common scenario but one I have seen in other systems frequently enough.

The IDE needs to be able to quickly find a unique project file based on the path of a `.dfy` file. You could have all project files end in the same suffix, like `*.dafny.toml` or `*.dfyp`, but what do you when the IDE finds two project files at the same distance of a `.dfy` file? You can prompt the user to ask which one should be used, but that seems needlessly complicated to me.

Reply:
Yes it has to be named `dafny.toml`, but that's so the IDE can easily find it, not because the project files are in the same CLI argument location as source files.